### DISCUSSION TITLE
Tuple expansions on multiple return values

### DISCUSSION THREAD

**Discussion by flysand7** [vlang/v#11890] [Ideas]
As known, V supports tuple returns and tuple assignments.

If, say, a function takes in a tuple argument, and another function returns a tuple, i should (?) be able to feed the output of one function into the input of the other like this:

```
fn sum(x int, y int)
  {
println(x+y)
  }

  fn prod_pair() (int, int)
  {
return 2,3
  }

  fn main()
  {
sum(prod_pair()) 
  }
```

## Current behaviour

```
    3 | fn a(x int, y int) {
main.v:12:2: error: expected 2 arguments, but got 1
   10 |
   11 | fn main() {
   12 |  sum(prod_pair())
      |  ~~~~~~~~~~~~~~
   13 | }
```

**Comment by dumblob** [comment]
I don't think this is the desired semantics. But with `...` syntax this should IMHO work (if it doesn't, then it's a bug if I'm not missing something subtle here).

**Comment by JalonSolov** [comment]
It looks like a bug.  `sum(<any one single thing>)` looks like you're passing a single argument.

In other words, it appears that V is not considering the return type of the function, it is simply counting the function as a single parameter.
